conformance/utils/kubernetes/helpers.go

```diff
@@ -499,6 +530,40 @@ func GatewayListenersMustHaveConditions(t *testing.T, client client.Client, time
 	require.NoErrorf(t, waitErr, "error waiting for Gateway status to have conditions matching expectations on all listeners")
 }
 
+// GatewayListenerMustHaveConditions checks if the given listener of the specified gateway has all
+// the specified conditions.
+func GatewayListenerMustHaveConditions(t *testing.T, client client.Client, timeoutConfig config.TimeoutConfig, gwName types.NamespacedName, listenerName string, conditions []metav1.Condition) {
+	t.Helper()
+
+	waitErr := wait.PollUntilContextTimeout(context.Background(), 1*time.Second, timeoutConfig.GatewayListenersMustHaveConditions, true, func(ctx context.Context) (bool, error) {
+		var listenerFound bool
+		var gw gatewayv1.Gateway
+		if err := client.Get(ctx, gwName, &gw); err != nil {
+			return false, fmt.Errorf("error fetching Gateway: %w", err)
+		}
+
+		for _, condition := range conditions {
+			for _, listener := range gw.Status.Listeners {
+				if listener.Name != gatewayv1.SectionName(listenerName) {
+					continue
+				}
+				listenerFound = true
```

Comment:
here is a bit of an example from above, if you have found the listener you are looking for you cannot simply exit the loop/break, as it will keep looking into conditions, right?